CASM-3505: Implement Argo WorkflowTemplate for ims-upload

jack-stanek-hpe opened the pull request:
This change adds a WorkflowTemplate which implements the ims-upload operation for the IUF. It handles setting up secrets, running the `load-ims-artifacts` image, and updating the product catalog with new IMS recipes and images.

Test Description:
Run workflow template with product distributions and mocked-out global_params, and ensure that IMS artifacts are uploaded successfully.

jack-stanek-hpe commented on workflows/iuf/operations/ims-upload/ims-upload-content-steps.yaml:
```diff
@@ -0,0 +1,196 @@
+
+    container:
+      # replace image with standard ims-load-artifacts once PR is merged
+      image: artifactory.algol60.net/csm-docker/unstable/cray-ims-load-artifacts:1.7.0-CASM-3506-adapt-for-iuf.19_33fd2e6
```
TODO: Change this to the latest `load-ims-artifacts` image when [the corresponding PR is merged](https://github.com/Cray-HPE/ims-load-artifacts/pull/29))

ekamin-hpe commented on workflows/iuf/operations/ims-upload/ims-upload.yaml:
```diff
@@ -0,0 +1,194 @@
+          - name: s3_credentials_secret_name
+            value: "{{steps.get-s3-secrets.outputs.parameters.secret_name}}"
+          - name: auth_token
+            value: "{{inputs.paramaters.auth_token}}"
```
```suggestion
            value: "{{inputs.parameters.auth_token}}"
```

jack-stanek-hpe commented:
I deleted the cruft changes from integration testing from `stages.yaml` and squashed the commits on the branch. The only content in this PR now is the `ims-upload` operation.

ekamin-hpe commented:
I added product catalog updates onto this branch, FYI @jack-stanek-hpe

ekamin-hpe commented on workflows/iuf/operations/ims-upload.yaml:
```diff
@@ -0,0 +1,224 @@
+        value: /etc/ssl/certs/ca-certificates.crt
+      - name: auth_token
+      - name: iuf_release_mount_path
+      - name: media_dir
```
I think this is no longer needed, along with ll. 51-52.

But then we do want to add `iuf_release_mount_path` as a volumeMount as we discussed in the meeting today.

ekamin-hpe replied:
Looks like `iuf_release_mount_path` is addressed but I still think you could remove `media_dir` as a parameter now.
